soh/src/code/z_map_exp.c

```diff
@@ -366,100 +353,49 @@ void Map_InitData(PlayState* play, s16 room) {
-                if ((!IS_RANDO && GET_EVENTCHKINF_CARPENTERS_FREE_ALL()) ||
-                    (IS_RANDO && CHECK_QUEST_ITEM(QUEST_GERUDO_CARD))) {
-                    extendedMapIndex = 0x17;
-                }
-            }
-            osSyncPrintf(VT_FGCOL(BLUE));
-            osSyncPrintf("ＫＫＫ＝%d\n", extendedMapIndex);
-            osSyncPrintf(VT_RST);
-            sEntranceIconMapIndex = extendedMapIndex;
-            //DmaMgr_SendRequest1(interfaceCtx->mapSegment,
-                                //(uintptr_t)_map_grand_staticSegmentRomStart + gMapData->owMinimapTexOffset[extendedMapIndex],
-                                //gMapData->owMinimapTexSize[mapIndex], __FILE__, __LINE__);
-
-            if (sEntranceIconMapIndex < 24) {
-                play->interfaceCtx.mapSegment[0] = ResourceGetDataByName(minimapTableOW[sEntranceIconMapIndex]);
-                play->interfaceCtx.mapSegmentName[0] = minimapTableOW[sEntranceIconMapIndex];
-            }
-
-            interfaceCtx->unk_258 = mapIndex;
-            break;
-        case SCENE_DEKU_TREE:
-        case SCENE_DODONGOS_CAVERN:
-        case SCENE_JABU_JABU:
-        case SCENE_FOREST_TEMPLE:
-        case SCENE_FIRE_TEMPLE:
-        case SCENE_WATER_TEMPLE:
-        case SCENE_SPIRIT_TEMPLE:
-        case SCENE_SHADOW_TEMPLE:
-        case SCENE_BOTTOM_OF_THE_WELL:
-        case SCENE_ICE_CAVERN:
-        case SCENE_DEKU_TREE_BOSS:
-        case SCENE_DODONGOS_CAVERN_BOSS:
-        case SCENE_JABU_JABU_BOSS:
-        case SCENE_FOREST_TEMPLE_BOSS:
-        case SCENE_FIRE_TEMPLE_BOSS:
-        case SCENE_WATER_TEMPLE_BOSS:
-        case SCENE_SPIRIT_TEMPLE_BOSS:
-        case SCENE_SHADOW_TEMPLE_BOSS:
-            osSyncPrintf(VT_FGCOL(YELLOW));
-            // "Deku Tree Dungeon MAP Texture DMA"
-            osSyncPrintf("デクの樹ダンジョンＭＡＰ テクスチャＤＭＡ(%x) scene_id_offset=%d  VREG(30)=%d\n", room,
-                         mapIndex, VREG(30));
-            osSyncPrintf(VT_RST);
-            //DmaMgr_SendRequest1(play->interfaceCtx.mapSegment,
-                                //(uintptr_t)_map_i_staticSegmentRomStart +
-                                    //((gMapData->dgnMinimapTexIndexOffset[mapIndex] + room) * 0xFF0),
-                                //0xFF0, __FILE__, __LINE__);
-
-            play->interfaceCtx.mapSegment[0] = ResourceGetDataByName(
-                    minimapTableDangeon[gMapData->dgnMinimapTexIndexOffset[mapIndex] + room]);
-            play->interfaceCtx.mapSegmentName[0] = minimapTableDangeon[gMapData->dgnMinimapTexIndexOffset[mapIndex] + room];
-            R_COMPASS_OFFSET_X = gMapData->roomCompassOffsetX[mapIndex][room];
-            R_COMPASS_OFFSET_Y = gMapData->roomCompassOffsetY[mapIndex][room];
-            Map_SetFloorPalettesData(play, VREG(30));
-            osSyncPrintf("ＭＡＰ 各階ＯＮチェック\n"); // "MAP Individual Floor ON Check"
-            break;
+    SceneDBEntry* entry = SceneDB_Retrieve(mapIndex);
+
+    if (SCENEDB_ISOVERWORLD(entry)) {
```

Comment:
This handles special cases for overworld minimaps (filling the lake, discovering ST entrance, etc.). I have reworked it a bit to handle the logic and have hard-coded the special values that were originally in a tables past the normal ends. I have elected not to have this data in the SceneDB because I felt like these were too much of a niche case. For mods, I think this will be best handled with a hook.